src/App.tsx

```diff
@@ -1,24 +1,217 @@
-/* eslint-disable max-len */
-/* eslint-disable jsx-a11y/control-has-associated-label */
-import React from 'react';
-import { UserWarning } from './UserWarning';
+import React, {
+  useState,
+  useEffect,
+  useMemo,
+  useCallback,
+} from 'react';
 
-const USER_ID = 0;
+import { TodoList } from './components/TodoList';
+import { TodoHeader } from './components/TodoHeader';
+import { ErrorNotification } from './components/ErrorNotification';
+
+import {
+  getTodos,
+  createTodo,
+  deleteTodo,
+  updateTodoStatus,
+  updateTodoTitle,
+} from './api/todos';
+
+import { Todo } from './types/Todo';
+import { TodoStatus } from './types/TodoStatus';
+import { TodoFooter } from './components/TodoFooter';
+
+import { getVisibleTodos } from './utils/helpres';
+
+const USER_ID = 10681;
 
 export const App: React.FC = () => {
-  if (!USER_ID) {
-    return <UserWarning />;
-  }
+  const [todos, setTodos] = useState<Todo[]>([]);
+  const [error, setError] = useState<string | null>(null);
+  const [filter, setFilter] = useState(TodoStatus.ALL);
+  const [tempTodo, setTempTodo] = useState<Todo | null>(null);
+  const [loadingTodos, setLoadingTodos] = useState([0]);
+
+  useEffect(() => {
+    getTodos(USER_ID)
+      .then((fetchedTodos: Todo[]) => {
+        setTodos(fetchedTodos);
+      })
+      .catch((fetchedError: Error) => {
+        setError(fetchedError?.message ?? 'Something went wrong');
+      });
+  }, []);
+
+  useEffect(() => {
+    let timeout: NodeJS.Timeout;
+
+    if (error) {
+      timeout = setTimeout(() => {
+        setError(null);
+      }, 3000);
+    }
+
+    return () => {
+      clearTimeout(timeout);
+    };
+  }, [error]);
+
+  const activeTodos = useMemo(() => (
+    todos.filter(todo => !todo.completed)
+  ), [todos]);
+
+  const completedTodos = useMemo(() => (
+    todos.filter(todo => todo.completed)
+  ), [todos]);
```

Comment:
Please, do not use useCallback / useMemo hooks until you see real problems with optimisation in your application. This is definitely not the case to use them. You pass this function to component, which is not even wrapped in memo, please, make sure to read more about useCallback / useMemo, when and how to appropriately use them